### ISSUE TITLE
migtool error

### ISSUE THREAD

**Issue by dk766** [IvS-KULeuven/PlatoSim3#307]
In reference to #300 
While I trying to migrate an older PlatoSim input configuration file to the PlatoSim v3.2.0-RC1-542-g60c8a16 I run into the following error:

    (env) chifuv@plato6:/data/plato/simulations/soft/default/python> python migtool.py /data/plato simulations/workspace/heller_20181002_144621/inputfiles/inputfile.yaml ./new_inputfile.yaml
    Traceback (most recent call last):
        File "migtool.py", line 408, in <module
        main()
    File "migtool.py", line 395, in main
        newDict= load_yaml(args.newFile)
     File "migtool.py", line 196, in load_yaml
        fileStream = open(filename)
    FileNotFoundError: [Errno 2] No such file or directory: './new_inputfile.yaml'

My environment:

    pip freeze
        astropy==3.0.4
        h5py==2.8.0
        numpy==1.15.1
        PyYAML==3.13
        six==1.11.0
    python --version
        Python 3.6.4 :: Anaconda, Inc.

The File I was trying to convert is attached
      [inputfile.txt](https://github.com/IvS-KULeuven/PlatoSim3/files/2491679/inputfile.txt)

**Comment by JorisDeRidder** [comment]
I suspect the input arguments of migtool may be confusing. The syntax is
```
python migtool.py inputfile_old.yaml inputfile_new.yaml
```
and the output is `inputfile_old_changed.yaml`. So, `inputfile_old.yaml` is an older version, `inputfile_new.yaml` is an existing more recent input file that you can find in the directory inputfiles. The latter yaml file will be used to check what is missing and what has become redundant in the former yaml file. 

Perhaps it would be better to put the name of the output file also an argument of migtool.py, which would clarify things better.

**Comment by rhuygen** [comment]
Another possibility could be that the ```inputfile.yaml``` from the distribution is picked when only one argument is given, i.e. 

```$ python migtool.py myInputFile.yaml```

This will compare ```myInputFile.yaml``` with the ```inputfile.yaml``` from the distribution.

Additionally, we could write the new output to the screen/console unless the ```-o <filename>``` option is used to write to the given output file.

**Comment by dk766** [comment]
Oh, now I understand now why the tool was complaining that inputfile_new.yaml does not exist.

So basically you need 3 parameters: oldConfig, referenceConfig (from the current PlatoSim/inputfiles) and newConfig (which is the oldConfig migrated to the new format). One way would be that the migtool.py to know where to find the referenceConfig (after all this is an internal configuration file) and the user only provides the oldConfig and (optionally) the name of the newConfig. If the referenceConfig cannot be found the tool should provide an error (and ideally an explanation).
This is what I was actually expecting to be the case in my attempt from above.

**Comment by rhuygen** [comment]
I propose to change the command interface as such:

```
[1] $ python migtool.py my_old_inputfile.yaml
[2] $ python migtool.py -o new_adapted_inputfile.yaml my_old_inputfile.yaml
```

[1] will write the new YAML info on the terminal (stdout) which can then be inspected and/or saved with redirection
[2] will write the new YAML info to the file specified with the ```-o``` option, i.e. ```new_adapted_inputfile.yaml```

Is that oké for you?

**Comment by dk766** [comment]
Yes, the above proposed options (and behaviour) would correct the original parameter confusion.

**Comment by rhuygen** [comment]
This has been implemented with #298.

**Comment by SaraRegibo** [comment]
Documented in pull requested #314 (locally in the `develop` branch only).
